File: CSharp/DataStructures/Lists/LinkedListNode.cs
namespace DataStructures.Lists
{
    internal class LinkedListNode<T>
    {
        public T NodeData { get; set; }

        public LinkedListNode<T>? NextNode { get; set; }

        public LinkedListNode<T>? PreviousNode { get; set; }

        public LinkedListNode(T data, LinkedListNode<T>? previous)
        {
            this.NodeData = data;
            this.PreviousNode = previous;
        }
    }
}

File: Java/src/Lists/LinkedListNode.java
package Lists;

class LinkedListNode<T> {
    public LinkedListNode<T> PreviousNode = null;
    public LinkedListNode<T> NextNode = null;
    public T Data = null;

    public LinkedListNode(T data, LinkedListNode<T> previousNode) {
        PreviousNode = previousNode;
        Data = data;
    }
}
